### ISSUE TITLE
When selecting a suggestion in HTML from the vscode intellisense, it deletes a few characters after the pointer

### ISSUE THREAD

**Issue by elazarza** [angular/vscode-ng-language-service#1791]
# 🐞 bug report

### Is this a regression?

No
### Description

When starting to type something in the template, for example: a class-name, and visual code suggests a completion, when selecting the completion while the extension is enabled the completion deletes 2-5 characters after the pointer. not happening when the extension is disabled. video included

## Bug Type
What does this bug affect

- [x ] Angular Language Service VSCode extension
- [ ] Angular Language Service server

## Reproduction

1. start typing anything that prompts a suggestion in the template in vscode, where the line goes on after that location
2. select that suggestion

**Expected behavior**

The suggestion is inserted into the editor, and the rest of the line remains.

### **Logs**

Log file gives us deep insight into the behavior and performance of the extension. If the issue is a performance
problem or an error occured, please provide the output of the log file below.

Set Angular Log level to verbose.

<pre><code>

</code></pre>

### **Screenshots**

![Example Screenshot](https://i.imgur.com/IP88LQi.gif)

## 🌍  Your Environment
**Angular Version:**
<pre>

<code>

Angular: 13.2.4
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, platform-browser, platform-browser-dynamic, router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1302.5
@angular-devkit/build-angular      13.2.5
@angular-devkit/core               13.2.5
@angular-devkit/schematics         13.2.5
@angular/cli                       13.2.5
@angular/material-moment-adapter   14.0.2
@schematics/angular                13.2.5
rxjs                               7.5.4
typescript                         4.5.5
</code></pre>

Extension Version:
<pre><code>

v14.2.0
</code></pre>

VSCode Version:
<pre><code>

Version: 1.72.2 (user setup)
Commit: d045a5eda657f4d7b676dedbfa7aab8207f8a075
Date: 2022-10-12T22:15:18.074Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
</code></pre>

Operating System:
<pre><code>

Windows 10
</code></pre>

Extension options:
<pre><code>

{
    "angular.enable-strict-mode-prompt": false
}
</code></pre>

**Anything else relevant?**

**Comment by atscott** [comment]
@elazarza I investigated this in the past and found it to be an issue in VSCode. I submitted https://github.com/microsoft/vscode/issues/132785 but I think it was misunderstood or didn't get enough attention to be investigated fully. Unfortunately, there's nothing that can be done on our side for this.

**Comment by atscott** [comment]
For reference: https://github.com/angular/vscode-ng-language-service/issues/1503#issuecomment-916160786

**Comment by elazarza** [comment]
> @elazarza I investigated this in the past and found it to be an issue in VSCode. I submitted [microsoft/vscode#132785](https://github.com/microsoft/vscode/issues/132785) but I think it was misunderstood or didn't get enough attention to be investigated fully. Unfortunately, there's nothing that can be done on our side for this.

Hi @atscott , thanks for the quick response.
Ok I guess we'll have to live with it then. I tried commenting on your submission to microsoft to gain attention but its locked of course.

thanks!

**Comment by d9j** [comment]
ANOTHER problem in the latest update is that when clicking autocompletion it is breaking line.. very annoying issue

**Comment by ahules** [comment]
I have the same problem. Just removes few characters after selection the suggestion

**Comment by LaXiS96** [comment]
Can this issue be worked around somehow? It's happening way too often and starting to get old on me...
I'm surprised so few people have noticed and reported this problem, it drives me crazy!

Can't we disable this extension's changes to the HTML attributes suggestions and let the VSCode built-in language support do its thing?

**Comment by atscott** [comment]
I’ll have to take another look at this one soon. After rereading the response on the VSCode repo, I may have initially misinterpreted it. After completion items are returned, the language server appears to use the _current_ `scriptInfo` to compute replacement ranges but maybe needs to use the snapshot from when the request was received.

**Comment by d9j** [comment]
after  many struggles with plugin  . i had to move on. switched to vscode natively supported react. glad i dont have to work on angular anymore

**Comment by atscott** [comment]
I gave this another look at couldn't reproduce it. It's still not clear to me why this would be caused by the Angular Language Service. In the reproduction's screencast, the completion item isn't even provided by Angular.

**Comment by atscott** [comment]
Indeed, I can reproduce this without the Angular extension installed. I have attempted to report this issue again on the VSCode repo: https://github.com/microsoft/vscode/issues/251013.

**Comment by atscott** [comment]
Closing, since I’ve confirmed without a doubt that this is an issue with VSCode and can reproduce without the angular language service. You can follow the issue above for updates. 

A few workarounds possible: 

- disable the problematic completions with the setting` "[html]": {
  "editor.suggest.showWords": false,
  }, `
- use insert mode for html with the setting `"editor.suggest.insertMode": "insert"` (insert is the global default but overridden for html. See https://github.com/microsoft/vscode/issues/183746#issuecomment-1627663938) 
- type slower…?
